### ISSUE TITLE
Builds fail to rollout to stage 

### ISSUE THREAD

**Issue by pbergene** [openshiftio/openshift.io#1532]
Looks like we have the return of https://github.com/openshiftio/openshift.io/issues/1254 or something very similar.

Steps to reproduce: Reset tenant, create new space, do vertx quickstart, promote build to run, then  attempt to run the build a second time.

Was this never fixed and rolled out to production a second time?

@rawlingsj @maxandersen @kbsingh @rajdavies @chmouel @joshuawilson 

Jenkins console:
Could not update commit status. Message: Server returned HTTP response code: 201, message: 'Created' for URL: https://api.github.com/repos/pbergeneOsiotest1/newspaceforbuilds/statuses/d07e9e3d59444e73ed8953b0fca7858efe17baa6

ERROR: Error during kubernetes apply: java.lang.RuntimeException: Failed to create ImageStream from pipeline.json. io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PUT at: https://kubernetes.default/apis/image.openshift.io/v1/namespaces/pbergene-osiotest1-stage/imagestreams/newspaceforbuilds. Message: ImageStream in version "v1" cannot be handled as a ImageStream: parsing time "" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "2006". Received status: Status(apiVersion=v1, code=400, details=null, kind=Status, message=ImageStream in version "v1" cannot be handled as a ImageStream: parsing time "" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "2006", metadata=ListMeta(resourceVersion=null, selfLink=null, additionalProperties={}), reason=BadRequest, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.api.Controller.onApplyError(Controller.java:1459)
	at io.fabric8.kubernetes.api.Controller.applyImageStream(Controller.java:940)
	at io.fabric8.kubernetes.api.Controller.applyEntity(Controller.java:268)
	at io.fabric8.kubernetes.api.Controller.apply(Controller.java:227)
	at io.fabric8.kubernetes.pipeline.devops.ApplyStepExecution.run(ApplyStepExecution.java:248)
	at io.fabric8.kubernetes.pipeline.devops.ApplyStepExecution.run(ApplyStepExecution.java:102)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousStepExecution.start(AbstractSynchronousStepExecution.java:42)
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:224)
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:150)
	at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:108)
	at sun.reflect.GeneratedMethodAccessor225.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
	at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:123)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:110)
	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:153)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:157)
	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
	at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:57)
	at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:109)
	at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:82)
	at sun.reflect.GeneratedMethodAccessor194.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
	at com.cloudbees.groovy.cps.impl.CollectionLiteralBlock$ContinuationImpl.dispatch(CollectionLiteralBlock.java:55)
	at com.cloudbees.groovy.cps.impl.CollectionLiteralBlock$ContinuationImpl.item(CollectionLiteralBlock.java:45)
	at sun.reflect.GeneratedMethodAccessor222.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
	at com.cloudbees.groovy.cps.impl.LocalVariableBlock$LocalVariable.get(LocalVariableBlock.java:39)
	at com.cloudbees.groovy.cps.LValueBlock$GetAdapter.receive(LValueBlock.java:30)
	at com.cloudbees.groovy.cps.impl.LocalVariableBlock.evalLValue(LocalVariableBlock.java:28)
	at com.cloudbees.groovy.cps.LValueBlock$BlockImpl.eval(LValueBlock.java:55)
	at com.cloudbees.groovy.cps.LValueBlock.eval(LValueBlock.java:16)
	at com.cloudbees.groovy.cps.Next.step(Next.java:83)
	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:173)
	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:162)
	at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:122)
	at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:261)
	at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:162)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:19)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable$1.call(SandboxContinuable.java:35)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable$1.call(SandboxContinuable.java:32)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:108)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:32)
	at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:174)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:330)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$100(CpsThreadGroup.java:82)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:242)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:230)
	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:64)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:112)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PUT at: https://kubernetes.default/apis/image.openshift.io/v1/namespaces/pbergene-osiotest1-stage/imagestreams/newspaceforbuilds. Message: ImageStream in version "v1" cannot be handled as a ImageStream: parsing time "" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "2006". Received status: Status(apiVersion=v1, code=400, details=null, kind=Status, message=ImageStream in version "v1" cannot be handled as a ImageStream: parsing time "" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "2006", metadata=ListMeta(resourceVersion=null, selfLink=null, additionalProperties={}), reason=BadRequest, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:470)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:409)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:379)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:343)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleReplace(OperationSupport.java:260)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleReplace(OperationSupport.java:242)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleReplace(BaseOperation.java:778)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation$2.apply(HasMetadataOperation.java:109)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation$2.apply(HasMetadataOperation.java:104)
	at io.fabric8.openshift.api.model.DoneableImageStream.done(DoneableImageStream.java:27)
	at io.fabric8.openshift.api.model.DoneableImageStream.done(DoneableImageStream.java:6)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation.replace(HasMetadataOperation.java:116)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation.replace(HasMetadataOperation.java:37)
	at io.fabric8.kubernetes.api.Controller.applyImageStream(Controller.java:936)
	... 68 more

Finished: FAILURE

**Comment by joshuawilson** [comment]
To fix it last time @aslakknutsen had rolled back to Jenkins 4.0.41 fabric8-services/fabric8-tenant@1cc44c6

**Comment by joshuawilson** [comment]
So if a new tenant was released and this wasn't fixed then we could have the issue again.

**Comment by ldimaggi** [comment]
We are now at 4.0.44 - trying to recreate this now.

**Comment by bartoszmajsak** [comment]
Fix was applied to https://github.com/jenkinsci/kubernetes-pipeline-plugin/pull/64/files - but is it in this version of jenkins... Not sure where this "version" lives.

**Comment by bartoszmajsak** [comment]
I am just wondering - as we rolled back to `4.0.41` intentionally how come https://github.com/fabric8-services/fabric8-tenant/pull/490 was automerged and made this problem reappear? Can we somehow prevent such automerges from happening?

**Comment by rawlingsj** [comment]
The fix is in here https://github.com/fabric8io/openshift-jenkins-s2i-config/pull/121

**Comment by rawlingsj** [comment]
We upgraded all the plugins yesterday for 4.0.44 but the kubernetes pipeline plugin didn't have the latest kubernetes client.  We noticed this earlier and fixed it with the commit above.  There should be a PR on fabric8 tenant shortly.

**Comment by rawlingsj** [comment]
> I am just wondering - as we rolled back to 4.0.41 intentionally how come fabric8-services/fabric8-tenant#490 was automerged and made this problem reappear? Can we somehow prevent such automerges from happening if we know the issue is not fixed yet?

Yes that was already done yesterday but after the 4.0.44 release so we are protected going forward. https://github.com/fabric8io/openshift-jenkins-s2i-config/commit/9b02d7aba1b9b9b137be73453a14661eaa115295

**Comment by rawlingsj** [comment]
4.0.45 is being released by fabric8-tenant now https://jenkins.cd.test.fabric8.io/job/fabric8-services/job/fabric8-tenant-jenkins/job/master/43/console

**Comment by bartoszmajsak** [comment]
> Yes that was already done yesterday but after the 4.0.44 release so we are protected going forward. fabric8io/openshift-jenkins-s2i-config@9b02d7a

I wasn't looking for a global switch really, but thanks for the pointer.

**Comment by kbsingh** [comment]
does rolling this out involve a cluster wide update of the tenant spaces ?

**Comment by rawlingsj** [comment]
Fix is now being released in the tenant service https://jenkins.cd.test.fabric8.io/job/fabric8-services/job/fabric8-tenant/job/master/161/console

Once this is done it is just the automated PR on https://github.com/openshiftio/saas-openshiftio and the pipeline will wait for centos CI to build the prod image.  Then when that PR merges it's just the ci centos promote to prod that needs to run for the new 4.0.45 version to be available in prod.

**Comment by rawlingsj** [comment]
> does rolling this out involve a cluster wide update of the tenant spaces ?

I'm not actually sure what happens here, I know there was some work on the tenant service doing upgrades when a new version is available but TBH I'm not that familiar with it.

**Comment by rawlingsj** [comment]
@xcoulon do you happen to know if the tenant service will automatically upgrade users jenkins tenant namespaces with the new version when it's available?

**Comment by kbsingh** [comment]
addeding area/user/tenant - @xcoulon is this something that we are able to execute ? ( do not execute it yet though, we will need to sync on service side )

**Comment by rawlingsj** [comment]
@kbsingh FYI the fabric8 pipelines have finished now and PR merged on saas-openshift https://github.com/openshiftio/saas-openshiftio/pull/505.  Over to the ci centos promote to prod job is that correct?  Reason I mention it is because I saw a note about disabling auto promotes.  This one here would need to be manually triggered I guess?

**Comment by kbsingh** [comment]
@rawlingsj pushing now

**Comment by rawlingsj** [comment]
The fix in 4.0.45 is now available in prod - there's still an open question about how / if the tenant service updates existing tenant namespaces, but a fresh reset environment will bring in the latest version so ready for verification.  

@pbergene did you want to verify and close this issue?

**Comment by aslakknutsen** [comment]
It does not as there are no automated tests to verify the behavior of jenkins/che.

**Comment by bartoszmajsak** [comment]
> It does not as there are no automated tests to verify the behavior of jenkins/che.

I thought we have some https://github.com/fabric8-jenkins/godog-jenkins bdd tests which are written for this very purpose? Are they executed somewhere @rajdavies @rawlingsj?

**Comment by kbsingh** [comment]
( we need a seperate issue for talking through the automated-testing side of things - there might even be some scope for the e2e tests to grow a bit more ) . 

In the mean time, we need a way to get this updated tenant rolled out across the board - we have taken a hit on the service side being marked degraded, if at all possible, I would very much like to use that opportunity to get this deployed cluster wide. However, lets go ahead and get this issue closed if the fix is in, and open new ones as needed.

**Comment by rawlingsj** [comment]
> It does not as there are no automated tests to verify the behavior of jenkins/che.
I thought we have some https://github.com/fabric8-jenkins/godog-jenkins bdd tests which are written for this very purpose? Are they executed somewhere @rajdavies @rawlingsj?

The bdd tests run on pull requests to the jenkins image https://github.com/fabric8io/openshift-jenkins-s2i-config/pull/121

**Comment by pbergene** [comment]
Did 3 consecutive build and works like a charm! Thanks for moving so fast @rawlingsj 

Closing the issue as to mark it as cleared.

**Comment by rawlingsj** [comment]
yay!  no problem - it's a team effort

**Comment by bartoszmajsak** [comment]
@rawlingsj can the team responsible for them add the scenario to cover this regression then? would be awesome to have that really.

**Comment by rawlingsj** [comment]
Yes that is the aim but as you can imagine the team is currently focused on KT as a priority ATM.

**Comment by bartoszmajsak** [comment]
Sure, I totally understand that. The reason why I asked my original question what that when I go to PR build on Jenkins I see `TODO run BDD tests on OpenShift` and no tests archived. 

https://jenkins.cd.test.fabric8.io/blue/organizations/jenkins/fabric8io-auto-rel%2Fopenshift-jenkins-s2i-config/detail/PR-121/2/pipeline/49
